app/org/maproulette/models/dal/TaskDAL.scala

```diff
@@ -1655,6 +1659,59 @@ class TaskDAL @Inject()(override val db: Database,
     }
   }
 
+  def generateMentionNotifications(fromUser:User, comment:Comment, taskId:Long)(implicit c:Connection=null) = {
```

Comment:
So this particular class is getting rather large and taking on a log of responsibility that should be moved. My plan is to refactor this class. However new functionality that could be abstracted out should be. The notifications are definitely something that should be abstracted out into a separate class. The DAL class should really only be related to access the object data from the data storage, and this has become very muddy over the years. So if possible can we move this to a separate service.

Reply:
I've extracted the notification-related code into new NotificationDAL and NotificationController classes, unburdening the TaskDAL, UserDAL, and UserController classes.